src/components/data-exploration/embedding/Embedding.jsx

```diff
@@ -132,23 +132,33 @@ const Embedding = (props) => {
   }, [focusedExpression]);
 
   useEffect(() => {
-    if (cellSetHierarchy) {
-      const mapping = cellSetHierarchy.reduce((keyToClusterNameMap, rootHierarchy) => {
-        if (rootHierarchy.children.length > 0) {
-          rootHierarchy.children.forEach((child) => {
-            // eslint-disable-next-line no-param-reassign
-            keyToClusterNameMap[child.key] = _.capitalize(rootHierarchy.key);
-          });
-        }
-        return keyToClusterNameMap;
-      }, {});
-      setClusterKeyToNameMap(mapping);
-    }
-
-    if (cellSetProperties) {
-      setCellSetClusters(Object.entries(cellSetProperties).filter(([, cellSet]) => cellSet.type === 'cellSets'));
+    if (selectedCell) {
+      const expressionToDispatch = focusedExpression
+        ? focusedExpression.rawExpression.expression[selectedCell] : undefined;
+
+      // getting the root nodes which are of type cellSets
+      const rootClusterNodes = cellSetHierarchy.map(({ key }) => key)
+        .filter((key) => cellSetProperties[key].type === 'cellSets');
```

Comment:
Can you use the selector `getCellSetsHierarchyByKeys` instead of this?

Reply:
`getCellSetsHierarchyByKeys` searches by keys, instead i need to get all the clusters that have `type: cellSet` (in this case louvain and scratchpad) for the tooltip

Reply:
Can you use `getCellSetsHierarchyByType` then?